doc/changes/latest.inc

```diff
@@ -298,3 +298,5 @@ API changes
 - ``mne.beamformer.tf_dics`` has been deprecated and will be removed in MNE-Python 0.24 (:gh:`9122` by `Britta Westner`_)
 
 - Fitting `~mne.preprocessing.ICA` on baseline-corrected `~mne.Epochs`, and / or applying it on baseline-corrected `~mne.Epochs` or `~mne.Evoked` data will now display a warning. Users are advised to only baseline correct their data after cleaning is completed (:gh:`9033` by `Richard Höchenberger`_)
+
+- `mne.preprocessing.compute_proj_eog` and `mne.preprocessing.compute_proj_ecg` now return empty lists if no EOG or ECG events, respectively, could be found. Previously, we'd return ``None`` in these situations, which does not match the documented behavior of returning a list of projectors (:gh:`9277` by `Richard Höchenberger`_)
```

Comment:
```suggestion
- :func:`mne.preprocessing.compute_proj_eog` and :func:`mne.preprocessing.compute_proj_ecg` now return empty lists if no EOG or ECG events, respectively, could be found. Previously, we'd return ``None`` in these situations, which does not match the documented behavior of returning a list of projectors (:gh:`9277` by `Richard Höchenberger`_)
```

Reply:
That's actually not required anymore since to some changes @drammock made to our doc building process a while ago :)

Reply:
I think this should go in BUG rather than API because it's a backward-incompatible change with no deprecation cycle

Reply:
> because it's a backward-incompatible change with no deprecation cycle

Yeah that's specifically why I put it into the API section …

Reply:
I think we went through this a bit already in https://github.com/mne-tools/mne-python/pull/9027#issuecomment-804348211 and https://github.com/mne-tools/mne-python/pull/8870#discussion_r577136501 but just to be more specific I think our current practice is:

1. "Enhancements": new, non-breaking code changes
2. "Bug": bugfixes, can immediately break existing code without any warning or deprecation cycle, but for a good reason (i.e., the existing code probably did not work correctly)
3. "API": changes to the API that do not break code right away but will probably do so in the next release, and almost always emit a deprecation warning to tell you

Should we add this to the developer guide so that it's explicitly written somewhere?

Reply:
Ok I will move this to "Bug", even though I believe since been present for so long it's expected behavior that we're changing, so as a user I would expect this in the "API" section – also considering that bug section is so long that it's impossible to read in its entirety …

Reply:
> for so long it's expected behavior now that we're changing

The point is not about the duration the bug has existed, but rather there is a release cycle to change your code (API) or if it breaks immediately (BUG)

> as a user I would expect this in the "API" section – also considering that the bug section is so long that it's impossible to read it in its entirety …

Again please open an issue about this

Reply:
> The point is not about the duration the bug has existed, but rather there is a release cycle to change your code (API) or if it breaks immediately (BUG)

Ok, got it!